### ISSUE TITLE
lmdb-java throws UnsatisfiedLinkError on M2 macOS

### ISSUE THREAD

**Issue by paul-snively** [lmdbjava/lmdbjava#227]
Thank you for the 0.9.0 release!

Unfortunately, one of our developers on Apple Silicon, when trying to use the new 0.9.0 dependency, gets:

rnode.bootstrap   | 2023-12-06 18:26:40,299 [main] ERROR coop.rchain.node.Main$ - Unhandled exception in thread main
rnode.bootstrap   | java.lang.UnsatisfiedLinkError: could not load FFI provider jnr.ffi.provider.jffi.Provider
rnode.bootstrap   |     at jnr.ffi.provider.InvalidProvider$1.loadLibrary(InvalidProvider.java:49)
rnode.bootstrap   |     at jnr.ffi.LibraryLoader.load(LibraryLoader.java:420)
rnode.bootstrap   |     at jnr.ffi.LibraryLoader.load(LibraryLoader.java:399)
rnode.bootstrap   |     at org.lmdbjava.Library.<clinit>(Library.java:81)
rnode.bootstrap   |     at org.lmdbjava.Env$Builder.open(Env.java:580)
I've manually downloaded the published .jars for jnr-ffi, jnr-jffi, and lmdbjava, extracted them, and honestly, everything looks in order. I've used the Ammonite REPL for Scala, imported lmdbjava from the central Maven repo, and walked through the steps of the tutorial to create a database, insert data, retrieve data, etc. successfully, but I'm on an X86_64 macOS box. Any advice would be welcome.

I feel somewhat bad about raising this here, as I strongly suspect this is an upstream JNR FFI/JFFI issue, but perhaps we can collaborate with the appropriate people toward a resolution.

Thanks again!

**Comment by paul-snively** [comment]
OK, one of my colleagues chased this down. It turns out the `.jar` file doesn't include the `lmdb` shared libraries for the platforms, but rather, it contains shared libraries for the supported platforms that, in turn, dynamically link to the `lmdb` shared library, which is then required to be found in one of the standard locations for the platform.

We can work around this for the moment by using `brew install lmdb` on macOS, the moral equivalent on various Linux distros, and by adding the appropriate stanza to our Dockerfile for building our Docker image, but that's more than we'd like to keep track of for onboarding, and seems to me to be against the intent of including the wrapper shared libraries in the `.jar` and having the `.jar` be self-contained.

In other words, I suspect this is just an oversight. :smile:

Thanks again for all the great work on lmdbjava!

**Comment by benalexau** [comment]
Unfortunately I do not have an Apple machine to reproduce this on. `ldd` of the Linux SO looks fine to me and it was auto build by Zig like the rest of the bundled libraries:

```
$ ls -l org/lmdbjava/*-*-*.*
-rwxr-xr-x 1 bpa bpa 323936 Dec  5 01:40 org/lmdbjava/aarch64-linux-gnu.so
-rwxr-xr-x 1 bpa bpa 349624 Dec  5 01:40 org/lmdbjava/aarch64-macos-none.so
-rwxr-xr-x 1 bpa bpa 327544 Dec  5 01:40 org/lmdbjava/x86_64-linux-gnu.so
-rwxr-xr-x 1 bpa bpa 303688 Dec  5 01:40 org/lmdbjava/x86_64-macos-none.so
-rwxr-xr-x 1 bpa bpa 246784 Dec  5 01:40 org/lmdbjava/x86_64-windows-gnu.dll

$ ldd org/lmdbjava/x86_64-linux-gnu.so 
	linux-vdso.so.1 (0x00007ffe80747000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f924e8b7000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f924e6d5000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f924e8f4000)
```

cc/ @wardle

**Comment by paul-snively** [comment]
I must apologize. This was shown to have been unrelated. I’ll close the issue. Sorry again to have wasted your time!

**Comment by benalexau** [comment]
Thanks for confirming and closing, @paul-snively.
